lib/circuitry/cli.rb

```diff
@@ -21,17 +23,24 @@ class CLI < Thor
       name <queue>-failures
     END
 
-    option :topics, aliases: :t, type: :array, required: :true
-    option :access_key, aliases: :a
-    option :secret_key, aliases: :s
-    option :dead_letter_queue, aliases: :d
-    option :region, aliases: :r
+    option :topic_names, aliases: :t, type: :array, required: true
+    option :access_key, aliases: :a, required: true
+    option :secret_key, aliases: :s, required: true
+    option :region, aliases: :r, required: true
+    option :dead_letter_queue_name, aliases: :d
+    option :visibility_timeout, aliases: :v, default: 30 * 60
+    option :max_receive_count, aliases: :n, default: 8
+
+    OPTIONS_KEYS_PUBLISHER_CONFIG = [:access_key, :secret_key, :region].freeze
+
+    OPTIONS_KEYS_SUBSCRIBER_CONFIG = [:access_key, :secret_key, :region, :dead_letter_queue_name,
+                                      :topic_names, :max_receive_count, :visibility_timeout].freeze
 
     def provision(queue_name)
-      with_custom_config(queue_name) do |config|
+      with_custom_config(queue_name) do
```

Comment:
If `with_custom_config` is no longer responsible for restoring the config, it doesn't really need to accept a block.  Maybe just change it to `#update_config` or something?